src/callbacks_step/analysis_surface_2d.jl

```diff
@@ -0,0 +1,294 @@
+    for boundary in indices
+        element = boundaries.neighbor_ids[boundary]
+        node_indices = boundaries.node_indices[boundary]
+        direction = indices2direction(node_indices)
+
+        i_node_start, i_node_step = index_to_start_step_2d(node_indices[1], index_range)
+        j_node_start, j_node_step = index_to_start_step_2d(node_indices[2], index_range)
+
+        i_node = i_node_start
+        j_node = j_node_start
+        for node_index in index_range
+            u_node = Trixi.get_node_vars(cache.boundaries.u, equations, dg, node_index,
+                                         boundary)
+
+            x = get_node_coords(node_coordinates, equations, dg, i_node, j_node,
+                                element)
+            # Extract normal direction at nodes which points from the elements outwards,
+            # i.e., *into* the structure.
+            normal_direction = get_normal_direction(direction, contravariant_vectors,
+                                                    i_node, j_node,
+                                                    element)
+
+            gradients_1 = get_node_vars(gradients_x, equations_parabolic, dg, i_node,
+                                        j_node, element)
+            gradients_2 = get_node_vars(gradients_y, equations_parabolic, dg, i_node,
+                                        j_node, element)
+
+            # Integral over whole boundary surface
+            value = variable(u_node, normal_direction, x, t, equations_parabolic,
+                             gradients_1, gradients_2)
+
+            coords[global_node_index, 1] = x[1]
+            coords[global_node_index, 2] = x[2]
+            values[global_node_index] = value
+            i_node += i_node_step
+            j_node += j_node_step
+            global_node_index += 1
+        end
+    end
+    # Sort with ascending x
+    sorting_indices = sortperm(coords[:, 1])
+    coords = coords[sorting_indices, :]
+    values = values[sorting_indices]
+
+    save_pointwise_file(surface_variable.output_directory, varname(variable), coords,
+                        values, t, iter)
+end
+
+varname(::Any) = @assert false "Surface variable name not assigned" # This makes sure default behaviour is not overwriting
+varname(pressure_coefficient::SurfacePressureCoefficient) = "CP_x"
+varname(friction_coefficient::SurfaceFrictionCoefficient) = "CF_x"
+
+function save_pointwise_file(output_directory, varname, coords, values, t, iter)
+    n_points = length(values)
+
+    filename = joinpath(output_directory, varname) * @sprintf("_%06d.h5", iter)
+
+    h5open(filename, "w") do file
+        # Add context information as attributes
+        attributes(file)["n_points"] = n_points
+        attributes(file)["variable_name"] = varname
+
+        file["time"] = t
+        file["timestep"] = iter
+        file["point_coordinates"] = coords
+        file["values"] = values
```

Comment:
```suggestion
        file["point_data"] = values
```
As a suggestion, to keep the same naming convention as for the time series output files?